helix-term/src/commands.rs

```diff
@@ -5519,6 +5520,24 @@ fn increment_impl(cx: &mut Context, increment_direction: IncrementDirection) {
     }
 }
 
+fn yank_diagnostic(cx: &mut Context) {
+    let (view, doc) = current_ref!(cx.editor);
+    let primary = doc.selection(view.id).primary();
+
+    let diag = doc
+        .diagnostics()
+        .iter()
+        .filter(|d| primary.overlaps(&helix_core::Range::new(d.range.start, d.range.end)))
+        .map(|d| d.message.clone());
+    let reg = cx.register.unwrap_or('+');
+    match cx.editor.registers.write(reg, diag.collect()) {
```

Comment:
Let's collect in the `let diag` above and record the length of the vec before `write` so we can improve the success message, like we have for `yank_impl`:

```rust
Ok(_) => cx.editor.set_status(format!(
    "Yanked {n} diagnostic{} to register {reg}",
    if n == 1 { "" } else { "s" }
)),
```

That should make it clearer when you're copying multiple diagnostics